src/System.CommandLine/Binding/ParameterDescriptor.cs

```diff
@@ -32,22 +32,16 @@ public bool AllowsNull
             {
                 if (_allowsNull is null)
                 {
-                    if (_parameterInfo.ParameterType.IsNullable())
-                    {
-                        _allowsNull = true;
-                    }
-
-                    if (_parameterInfo.HasDefaultValue &&
-                        _parameterInfo.DefaultValue is null)
-                    {
-                        _allowsNull = true;
-                    }
+                    _allowsNull = CalculateAllowsNull(_parameterInfo);
                 }
-
                 return _allowsNull ?? false;
             }
         }
 
+        public static bool CalculateAllowsNull(ParameterInfo parameterInfo) 
```

Comment:
```suggestion
        public static bool GetAllowsNull(ParameterInfo parameterInfo) 
```

Reply:
Or even just `AllowsNull`?

Reply:
I believe that conflicts with the property of the same name above.